protocol/lavaprotocol/finalization_consensus.go

```diff
@@ -253,16 +271,30 @@ func (fc *FinalizationConsensus) ExpectedBlockHeight(chainParser chainlib.ChainP
 		}
 		return lavaslices.Median(data)
 	}
+
 	medianOfExpectedBlocks := median(mapExpectedBlockHeights)
 	providersMedianOfLatestBlock := medianOfExpectedBlocks + int64(blockDistanceForFinalizedData)
 	if debug {
-		utils.LavaFormatDebug("finalization information", utils.Attribute{Key: "specId", Value: fc.specId}, utils.Attribute{Key: "mapExpectedBlockHeights", Value: mapExpectedBlockHeights}, utils.Attribute{Key: "medianOfExpectedBlocks", Value: medianOfExpectedBlocks}, utils.Attribute{Key: "latestBlock", Value: fc.latestBlockByMedian}, utils.Attribute{Key: "providersMedianOfLatestBlock", Value: providersMedianOfLatestBlock})
+		utils.LavaFormatDebug("finalization information",
+			utils.LogAttr("specId", fc.specId),
+			utils.LogAttr("mapExpectedBlockHeights", mapExpectedBlockHeights),
+			utils.LogAttr("medianOfExpectedBlocks", medianOfExpectedBlocks),
+			utils.LogAttr("latestBlock", fc.prevLatestBlockByMedian),
+			utils.LogAttr("providersMedianOfLatestBlock", providersMedianOfLatestBlock),
+		)
 	}
-	if medianOfExpectedBlocks > 0 && uint64(providersMedianOfLatestBlock) > fc.latestBlockByMedian {
-		if uint64(providersMedianOfLatestBlock) > fc.latestBlockByMedian+1000 && fc.latestBlockByMedian > 0 {
-			utils.LavaFormatError("uncontinuous jump in finalization data", nil, utils.Attribute{Key: "specId", Value: fc.specId}, utils.Attribute{Key: "s.prevEpochProviderHashesConsensus", Value: fc.prevEpochProviderHashesConsensus}, utils.Attribute{Key: "s.currentProviderHashesConsensus", Value: fc.currentProviderHashesConsensus}, utils.Attribute{Key: "latestBlock", Value: fc.latestBlockByMedian}, utils.Attribute{Key: "providersMedianOfLatestBlock", Value: providersMedianOfLatestBlock})
+
+	if medianOfExpectedBlocks > 0 && uint64(providersMedianOfLatestBlock) > fc.prevLatestBlockByMedian {
+		if uint64(providersMedianOfLatestBlock) > fc.prevLatestBlockByMedian+1000 && fc.prevLatestBlockByMedian > 0 {
+			utils.LavaFormatError("uncontinuous jump in finalization data", nil,
```

Comment:
This code was here before me. But my guess is that we still want to update the `fc.prevLatestBlockByMedian` to be the new `providersMedianOfLatestBlock` for future calculations.

Reply:
For testing purposes, I put a return statement right after the error and all the unit tests passed. So we're definitely missing a unit test here. Can you verify with Omer/Ran that only printing the error is intentional (and write a unit test if needed)?

Reply:
This part of the code was not written by me, but I'll check this one with @ranlavanet

Reply:
yes this is used for debugging at the moment no other solution. this could be explained if a provider was stuck for a long period of time. we dont punish it.